### ISSUE TITLE
[BUG] Zwift play not passing through to Zwift and the Kickr not showing up 

### ISSUE THREAD

**Issue by Arcendiss** [cagnulein/qdomyos-zwift#2911]
Set up:
QZ app running on pixel 9 pro XL, signed into beta, running version 2.18.9
Tacx Neo
Zwift play controllers fully updated, both firmware 1.3.1
Running Zwift on Windows 10 PC with a bluetooth dongle right under the trainer.

Open QZ app and it connects to trainer and zwift play just fine, play controllers are changing gear in the QZ screen
Open Zwift, I can connect phone (renamed to "PiXL" for bluetooth name length) perfectly for power, resistance and cadence
Zwift play controllers do not show up in the controllers menu
Going to power source, the kickr doesn't show up, PiXL is the only option there.

**Expected behavior**
According to your video "Zwift Virtual Gearing for all the trainers !", play controllers should be displayed in the controller section when QZ is connected as the phone and then I should switch over to the kickr trainer for the power to run in game.

[debug-Mon_Dec_16_19_16_28_2024.log](https://github.com/user-attachments/files/18156176/debug-Mon_Dec_16_19_16_28_2024.log)

**Comment by cagnulein** [comment]
Hi, you need to use the JUSTO name otherwise rouvy it doesn't detect it as
a smart trainer
Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 16 dic 2024 alle 20:50 Arcendiss ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
Hi, I guess you have some window firewall issue or some wifi router is filtering the frames from qz

Do an experiment : enable the android document icon setting in qz under the experimental settings and restart qz use zwift on the same android.

If you see the wahoo kickr virtual device it means that I was right, there is something on the network / pc

Let me know!

**Comment by Arcendiss** [comment]
I have the stock BT Smart Hub 2 if that information is of any use to you.

Forgive my stupid but can't find the setting you're talking about. In settings>experimental features?
The only Android related options I can see are:
WakeLock
Notification
Force Documents/QZ Folder

**Comment by cagnulein** [comment]
Here ![image](https://github.com/user-attachments/assets/09d21e62-b901-46ff-8bf1-a5a8db02447e)

**Comment by Arcendiss** [comment]
Ah ha, I understand, I see what you're trying to do. No, running zwift on the same phone shows absolutely nothing in the available power sources I'm afraid.

However, while we're experimenting I installed QZ on my old Samsung S10. There are icons on the menu show up on that phone don't on the Pixel which is making me think something incompatible on Pixel:
S10
![S10 qz menu](https://github.com/user-attachments/assets/6f786242-ed01-4c5a-8651-98849ffbb600)
Pixel
![pixel qz screenshot](https://github.com/user-attachments/assets/2e26943c-0c72-4f66-b525-7845c39e7d3f)

QZ on S10 will send Kickr to Zwift but still isn't passing through the Play controllers. Do you want log files from S10?

**Comment by cagnulein** [comment]
> Ah ha, I understand, I see what you're trying to do. No, running zwift on the same phone shows absolutely nothing in the available power sources I'm afraid.

Did you see the notification icon of qz?

It must work on that scenario, so can you show me the log again of this

Don't mind about the icons, it's because some phones don't handle uft8 icons 

About the s10 did you try with zwift on another device or one device directly?

**Comment by Arcendiss** [comment]
Yes, QZ notification was in the top bar. Screenshot showing QZ icon and no power sources:

![Qz icon showing no power](https://github.com/user-attachments/assets/8c19b20d-c513-410f-8ed9-165b2b0fe68f)

S10 was running QZ app and Zwift on my windows10 PC could detect the Kickr but still couldn't see Play controllers.

Is it something security turned on in the Pixel OS? Or something in latest Android? Pixel is on Android 15, I haven't got S10 on me at the moment so I'll let you know what OS that's using when I get home.

**Comment by cagnulein** [comment]
i have a pixel 6 and a pixel 8 both with android 15 and it works fine with zwift on the same device of qz. share me the debug log from the pixel, i just need to check that the dircon engine is up and running.



> S10 was running QZ app and Zwift on my windows10 PC could detect the Kickr but still couldn't see Play controllers.

zwift don't see the controller? are you sure the controller are blinking?

**Comment by Arcendiss** [comment]
Pixel:
[debug-Thu_Dec_19_17_39_48_2024.log](https://github.com/user-attachments/files/18201618/debug-Thu_Dec_19_17_39_48_2024.log)

Finds play controllers, finds trainer
Play controllers are changing gear in the QZ UI no problem at all, they are not blinking, lights are solid blue I assume because they're happy they're connected to QZ?
Zwift on PC: can see PiXL for power, resistance, cadence. Cannot see the Kickr. Cannot see the play controllers.
Zwift on Pixel: cannot see the kickr, cannot see the play controllers.

S10:
[debug-Thu_Dec_19_18_04_12_2024.log](https://github.com/user-attachments/files/18201745/debug-Thu_Dec_19_18_04_12_2024.log)

Finds play controllers, finds trainer
Play controllers are changing gear in the QZ UI. Play controllers are not blinking, solid blue.
Zwift on PC: can see S10 for power, resistance, cadence. Can see the Kickr for power, resistance, cadence. Cannot see the play controllers
For giggles: Zwift on Pixel can see S10 for power, resistance, cadence. Can see the Kickr for power, resistance, cadence. Cannot see the play controllers

**Comment by cagnulein** [comment]
You have to turn off the zwift play setting under the accessories settings
to be able to pair them on Zwift

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 19 dic 2024 alle 19:12 Arcendiss ***@***.***>
ha scritto:

**Comment by Arcendiss** [comment]
But if I do that it doesn't connect to QZ and doesn't control the virtual gears?

**Comment by cagnulein** [comment]
Then qz will get the gears from zwift directly! It's the power of qz :)
